internal/command/migrate_to_v2/migrate_to_v2.go

```diff
@@ -682,6 +979,7 @@ func (m *v2PlatformMigrator) resolveMachineFromAlloc(alloc *api.AllocationStatus
 	launchInput.Config.Image = m.img
 	launchInput.Config.Env = lo.Assign(m.appConfig.Env)
 	if m.appConfig.PrimaryRegion != "" {
+		m.primaryRegion = m.appConfig.PrimaryRegion
```

Comment:
Nit: I feel like this spiritually belongs in determinePrimaryRegion
```diff
func (m *v2PlatformMigrator) determinePrimaryRegion(ctx context.Context) error {
+    defer func(){
+        m.primaryRegion = m.appConfig.PrimaryRegion
+    }()
```

Only mentioning because this only runs once machines are being created, and that might lead to a confusing dependency chain if something tries to use `m.primaryRegion` before machines are created